sources/osgWrappers/serializers/osgAnimation.js

```diff
@@ -394,25 +400,105 @@ define( [
         var rigPromise = osgWrapper.Geometry( input, rigGeom );
         rigGeom._boneNameID = jsonObj.BoneMap;
 
+        var mergeGeometry = function ( from, to ) {
+            // Merge rigGeometry and sourceGeometry, we keep source instance and adds
+            // some references to the rigGeometry
+
+            //Merge primitives
+            to.primitives = from.primitives;
+
+            //Merge Attributes
+            var keys = Object.keys( from.attributes );
+            for ( var i = 0, l = keys.length; i < l; i++ ) {
+                var key = keys[ i ];
+                to.attributes[ key ] = from.attributes[ key ];
+            }
+
+            //Merge state set
+            to.stateset = from.stateset;
+
+            //@TODO Think to merge more stuff here
+        };
+
         //Import source geometry and merge it with the rigGeometry
         var sourceGeometry = jsonObj.SourceGeometry[ 'osg.Geometry' ];
         var geomPromise;
         if ( sourceGeometry ) {
             input.setJSON( sourceGeometry );
-            geomPromise = osgWrapper.Geometry( input, rigGeom );
+            rigGeom.setSourceGeometry( new Geometry() );
+            geomPromise = osgWrapper.Geometry( input, rigGeom.getSourceGeometry() );
+        } else {
+            sourceGeometry = jsonObj.SourceGeometry[ 'osgAnimation.MorphGeometry' ];
+            if ( sourceGeometry ) {
+                input.setJSON( sourceGeometry );
+                rigGeom.setSourceGeometry( new MorphGeometry() );
+                geomPromise = osgAnimationWrapper.MorphGeometry( input, rigGeom.getSourceGeometry() );
+            } else {
+                Notify.warn( 'SourceGeometry type no recognized' );
+            }
         }
 
-        return P.all( [ rigPromise, geomPromise ] ).then( function () {
+        return P.all( [ rigPromise, geomPromise ] ).then( function ( promises ) {
+            mergeGeometry( promises[ 0 ].getSourceGeometry(), promises[ 0 ] );
             return rigGeom;
         } );
     };
 
+    osgAnimationWrapper.MorphGeometry = function ( input, morphGeom ) {
+        var jsonObj = input.getJSON();
+
+        if ( /*!jsonObj.Name ||*/ !jsonObj.MorphTargets )
+            return P.reject();
+
+        var morphTargets = jsonObj.MorphTargets;
+        var arrayPromise = [];
+
+        arrayPromise.push( osgWrapper.Geometry( input, morphGeom ) );
+
+        for ( var i = 0, l = morphTargets.length; i < l && i < 4; i++ ) {
```

Comment:
keeps loop simple not condition in the i < l
adds conditions in the loop if you need